doc/releases/changelog-dev.md

```diff
@@ -4,7 +4,74 @@
 
 <h3>New features since last release</h3>
 
-* A tensor network templates module has been added. Quantum circuits with the shape of a matrix product state tensor network can now be easily implemented. Motivation and theory can be found in [arXiv:1803.11537](https://arxiv.org/abs/1803.11537). [(#1871)](https://github.com/PennyLaneAI/pennylane/pull/1871)
+* The `RotosolveOptimizer` has been generalized to arbitrary frequency spectra
+  in the cost function.
+  [(#2081)](https://github.com/PennyLaneAI/pennylane/pull/2081)
+
+  Previously, the `RotsolveOptimizer` was available for cost functions with
+  frequency spectra that only contained integers, and the maximal frequency
+  (instead of the number of frequencies) determined the cost of the optimization.
+  Now arbitrary frequencies are supported.
+
+  Consider the QNode
+  ```python
+  dev = qml.device("default.qubit", wires=2)
+
+  @qml.qnode(dev)
+  def qnode(x, Y):
+      qml.RX(2.5*x, wires=0)
+      qml.CNOT(wires=[0, 1])
+      qml.RZ(0.3*Y[0], wires=0)
+      qml.CRY(1.1*Y[1], wires=[1, 0])
+      return qml.expval(qml.PauliX(0)@qml.PauliZ(1))
+
+  x = np.array(0.8, requires_grad=True)
+  Y = np.array([-0.2, 1.5], requires_grad=True)
+  ```
+
+  Its frequency spectra can be easily obtained via `qml.fourier.qnode_spectrum`:
+  ```pycon
+  >>> spectra = qml.fourier.qnode_spectrum(qnode)(x, Y)
+  >>> spectra
+  {'x': {(): [-2.5, 0.0, 2.5]},
+   'Y': {(0,): [-0.3, 0.0, 0.3], (1,): [-1.1, -0.55, 0.0, 0.55, 1.1]}}
+  ```
+
+  We may then initialize the `RotosolveOptimizer` and minimize the QNode cost function
+  by providing this information about the frequency spectra. We also compare the cost at
+  each step to the initial cost.
+  ```pycon
+  >>> cost_init = qnode(x, Y)
+  >>> opt = qml.RotosolveOptimizer()
+  >>> for _ in range(2):
+  ...     x, Y = opt.step(qnode, x, Y, spectra=spectra)
```

Comment:
I am partly wondering why the spectra is not passed an initialization. I suppose it makes more sense to do this:

```python
>>> opt = qml.RotosolveOptimizer(qnode, spectra=spectra)
>>> for _ in range(2):
...     x, Y = opt.step(x, Y)
```

but this is a longer term breaking change, and connects to #1564